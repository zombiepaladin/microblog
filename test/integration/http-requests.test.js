const request = require('request');
const server = require('../../src/server');

const PORT = 3000;
const HOST = `http://localhost:${PORT}`;

// This hook is executed before any tests are run
beforeAll(function(){
  server.start(PORT);
});

// This hook is executed after all tests have been run
afterAll(function(){
  server.stop();
});

test('Serve blog page', function(done) {
  request(HOST + '/', function(err, res, body) {
    expect(err).toBeNull();
    expect(body).toMatchSnapshot();
    done();
  });
});

test('Create blog post', function(done) {
  request.post(HOST + '/', {form: {message: "test"}}, function(err, res, body) {
    expect(err).toBeNull();
    expect(body).toMatchSnapshot();
    done();
  });
});
