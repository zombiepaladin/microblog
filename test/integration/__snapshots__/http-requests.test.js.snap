// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create blog post 1`] = `
"
    <!doctype html>
    <html>
      <head>
        <title>Microblog Example</title>
        <link type=\\"text/css\\" rel=\\"stylesheet\\" href=\\"style.css\\"></link>
      </head>
      <div class=\\"column\\">
        <h1>Microblog</h1>
        <div id=\\"blog\\">
          <div class=\\"post\\">test</div>
        </div>
        <form id=\\"new-post\\" method=\\"POST\\">
          <fieldset>
            <textarea id=\\"message\\" name=\\"message\\" cols=\\"5\\"></textarea>
            <input type=\\"submit\\" value=\\"Post\\"/>
          </fieldset>
        </form>
      </div>
    </body>
  </html>
  "
`;

exports[`Serve blog page 1`] = `
"
    <!doctype html>
    <html>
      <head>
        <title>Microblog Example</title>
        <link type=\\"text/css\\" rel=\\"stylesheet\\" href=\\"style.css\\"></link>
      </head>
      <div class=\\"column\\">
        <h1>Microblog</h1>
        <div id=\\"blog\\">
          
        </div>
        <form id=\\"new-post\\" method=\\"POST\\">
          <fieldset>
            <textarea id=\\"message\\" name=\\"message\\" cols=\\"5\\"></textarea>
            <input type=\\"submit\\" value=\\"Post\\"/>
          </fieldset>
        </form>
      </div>
    </body>
  </html>
  "
`;
